addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -983,3 +1004,196 @@ def remove_unfinished_work(self):
+            raise ValidationError(
+                _(
+                    "This type of operation cannot handle unpickable items. "
+                    "Please, contact your team leader to resolve the issue. "
+                    "Press back when resolved."
+                )
+            )
+
+        product = Product.get_or_create(product_id) if product_id else None
+        location = Location.get_or_create(location_id) if location_id else None
+        package = Package.get_or_create(package_name) if package_name else None
+        allow_partial = False
+        move_lines = self.get_available_move_lines()
+
+        if product:
+            if not location:
+                raise ValidationError(
+                    _("Missing location parameter for unpickable product %s.") % product.name
+                )
+
+            move_lines = move_lines.filtered(
+                lambda ml: ml.product_id == product and ml.location_id == location
+            )
+
+            msg = _("Unpickable product %s at location %s") % (product.name, location.name)
+
+            if lot_name:
+                lot_name = [lot_name] if isinstance(lot_name, str) else lot_name
+                move_lines = move_lines.filtered(lambda ml: ml.lot_id.name in lot_name)
+                msg += _(" with serial number %s") % lot_name
+
+            if package:
+                move_lines = move_lines.filtered(lambda ml: ml.package_id == package)
+                msg += _(" in package %s") % package.name
+            elif move_lines.package_id:
+                raise ValidationError(
+                    _("Unpickable product from a package but no package name provided.")
+                )
+
+            # at this point we should only have one move_line
+            quants = move_lines.get_quants()
+            allow_partial = True
+        elif package:
+            if not location:
+                location = package.location_id
+
+            move_lines = move_lines.filtered(
+                lambda ml: ml.package_id == package or ml.package_id.parent_id == package
+            )
+            quants = package._get_contained_quants()
+            msg = _("Unpickable package %s at location %s") % (package.name, location.name)
+        else:
+            raise ValidationError(
+                _("Missing required information for unpickable item: product or package.")
+            )
+
+        # Filter out the move lines already picked in case under received
+        move_lines = move_lines.filtered(lambda ml: ml.qty_done == 0)
+
+        if not move_lines:
+            raise ValidationError(
+                _("Cannot find move lines todo for unpickable item in this batch.")
+            )
+
+        pickings = move_lines.picking_id
+        original_picking_ids = {}
+        if raise_stock_investigation:
+            to_investigate = Picking.browse()
+        for picking in pickings:
+            picking.message_post(body=msg)
```

Comment:
Do we want to do this after the validations?

Reply:
I'm not sure I follow what you are suggesting. Can you expand please?